frontend/src/modules/contributor/components/edit/contributor-edit-profile-photo.vue

```diff
@@ -0,0 +1,134 @@
+            >
+              <template #suffix>
+                <lf-icon
+                  v-if="form.profilePhoto"
+                  name="close-circle-line"
+                  :size="20"
+                  class="text-gray-300 cursor-pointer"
+                  @click="form.profilePhoto = ''"
+                />
+              </template>
+            </lf-input>
+            <lf-field-messages
+              :validation="$v.profilePhoto"
+              :error-messages="{
+                url: 'This URL address is not valid',
+              }"
+            />
+            <lf-field-message v-if="!$v.profilePhoto.$invalid" type="hint">
+              Choose a 1:1 aspect ratio (square) to better fit your image.
+            </lf-field-message>
+          </lf-field>
+        </div>
+      </div>
+      <div class="px-6 py-4 gap-4 flex items-center justify-end">
+        <lf-button type="secondary-ghost" @click="close">
+          Cancel
+        </lf-button>
+        <lf-button
+          type="primary"
+          :disabled="$v.$invalid || defaultAvatar === form.profilePhoto"
+          :loading="sending"
+          @click="update()"
+        >
+          Update avatar
+        </lf-button>
+      </div>
+    </template>
+  </lf-modal>
+</template>
+
+<script setup lang="ts">
+import { Contributor } from '@/modules/contributor/types/Contributor';
+import LfAvatar from '@/ui-kit/avatar/Avatar.vue';
+import useContributorHelpers from '@/modules/contributor/helpers/contributor.helpers';
+import LfModal from '@/ui-kit/modal/Modal.vue';
+import { computed, reactive, ref } from 'vue';
+import LfButton from '@/ui-kit/button/Button.vue';
+import { url } from '@vuelidate/validators';
+import useVuelidate from '@vuelidate/core';
+import LfInput from '@/ui-kit/input/Input.vue';
+import { useContributorStore } from '@/modules/contributor/store/contributor.store';
+import LfField from '@/ui-kit/field/Field.vue';
+import LfFieldMessage from '@/ui-kit/field-message/FieldMessage.vue';
+import LfFieldMessages from '@/ui-kit/field-messages/FieldMessages.vue';
+import Message from '@/shared/message/message';
+import LfIcon from '@/ui-kit/icon/Icon.vue';
+
+const props = defineProps<{
+  modelValue: boolean,
+  contributor: Contributor,
+}>();
+
+const emit = defineEmits<{(e: 'update:modelValue', value: boolean): any}>();
+
+const { avatar } = useContributorHelpers();
+const { updateContributor } = useContributorStore();
+
+const defaultAvatar = avatar(props.contributor);
+
+const form = reactive({
+  profilePhoto: avatar(props.contributor),
+});
+
+const rules = {
+  profilePhoto: {
+    url,
+  },
+};
+const $v = useVuelidate(rules, form);
+
+const sending = ref<boolean>(false);
+
+const update = () => {
+  sending.value = true;
+  updateContributor(props.contributor.id, {
+    attributes: {
+      ...props.contributor.attributes,
+      avatarUrl: {
+        default: form.profilePhoto,
+      },
+    },
+  })
```

Comment:
I don't think this is ok, because this way we would be deleting all previous data we have for this attribute.
For instance, when we get a contributor from gitHub, this is what we have
```
attributes: {
   avatarUrl: {
      default: "https://...",
      gitHub: "https://..."
   }
}
```
When user updates the avatar, it should update to:
``` 
attributes: {
   avatarUrl: {
      default: "new url",
      custom: "new url",
      gitHub: "https://..."
   }
}

In code:
    attributes: {
      ...props.contributor.attributes,
      avatarUrl: {
        ...props.contributor.attributes.avatarUrl,
        default: form.profilePhoto,
        custom: form.profilePhoto,
      },
    },
```